File: .NET/VS2010TrainingKit/Labs/02 - ASP.NET/Source/Starting Point/C#/CustomerService.Model/Entities/OperationStatus.cs
// ----------------------------------------------------------------------------------
// Microsoft Developer & Platform Evangelism
// 
// Copyright (c) Microsoft Corporation. All rights reserved.
// 
// THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
// EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
// OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
// ----------------------------------------------------------------------------------
// The example companies, organizations, products, domain names,
// e-mail addresses, logos, people, places, and events depicted
// herein are fictitious.  No association with any real company,
// organization, product, domain name, email address, logo, person,
// places, or events is intended or should be inferred.
// ----------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Diagnostics;

namespace CustomersService.Model.Entities
{
    [DebuggerDisplay("Status: {Status}")]
    public class OperationStatus
    {
        public bool Status { get; set; }
        public int RecordsAffected { get; set; }
        public string Message { get; set; }
        public Object OperationID { get; set; }
        public string ExceptionMessage { get; set; }
        public string ExceptionStackTrace { get; set; }
        public string ExceptionInnerMessage { get; set; }
        public string ExceptionInnerStackTrace { get; set; }

        public static OperationStatus CreateFromException(string message, Exception ex)
        {
            OperationStatus opStatus = new OperationStatus
            {
                Status = false,
                Message = message,
                OperationID = null
            };

            if (ex != null)
            {
                opStatus.ExceptionMessage = ex.Message;
                opStatus.ExceptionStackTrace = ex.StackTrace;
                opStatus.ExceptionInnerMessage = (ex.InnerException == null) ? null : ex.InnerException.Message;
                opStatus.ExceptionInnerStackTrace = (ex.InnerException == null) ? null : ex.InnerException.StackTrace;
            }
            return opStatus;
        }
    }
}

File: .NET/VS2010TrainingKit/Labs/02 - ASP.NET/Source/Starting Point/VB/CustomerService.Model/Entities/OperationStatus.vb
' ----------------------------------------------------------------------------------
' Microsoft Developer & Platform Evangelism
' 
' Copyright (c) Microsoft Corporation. All rights reserved.
' 
' THIS CODE AND INFORMATION ARE PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, 
' EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES 
' OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
' ----------------------------------------------------------------------------------
' The example companies, organizations, products, domain names,
' e-mail addresses, logos, people, places, and events depicted
' herein are fictitious.  No association with any real company,
' organization, product, domain name, email address, logo, person,
' places, or events is intended or should be inferred.
' ----------------------------------------------------------------------------------

Imports System.Text

Namespace CustomersService.Model.Entities
	<DebuggerDisplay("Status: {Status}")>
	Public Class OperationStatus
		Public Property Status() As Boolean
		Public Property RecordsAffected() As Integer
		Public Property Message() As String
		Public Property OperationID() As Object
		Public Property ExceptionMessage() As String
		Public Property ExceptionStackTrace() As String
		Public Property ExceptionInnerMessage() As String
		Public Property ExceptionInnerStackTrace() As String

		Public Shared Function CreateFromException(ByVal message As String, ByVal ex As Exception) As OperationStatus
			Dim opStatus As OperationStatus = New OperationStatus With {.Status = False, .Message = message, .OperationID = Nothing}

			If ex IsNot Nothing Then
				opStatus.ExceptionMessage = ex.Message
				opStatus.ExceptionStackTrace = ex.StackTrace
				opStatus.ExceptionInnerMessage = If(ex.InnerException Is Nothing, Nothing, ex.InnerException.Message)
				opStatus.ExceptionInnerStackTrace = If(ex.InnerException Is Nothing, Nothing, ex.InnerException.StackTrace)
			End If
			Return opStatus
		End Function
	End Class
End Namespace
